composer/callbacks/mlperf.py

```diff
@@ -0,0 +1,314 @@
+
+        dist.barrier()
+        if rank_zero():
+            self.mllogger.event(key=constants.RUN_START)
+
+    def epoch_start(self, state: State, logger: Logger) -> None:
+        if rank_zero():
+            self.mllogger.event(key=constants.EPOCH_START, metadata={'epoch_num': state.timer.epoch.value})
+            self.mllogger.event(key=constants.BLOCK_START,
+                                metadata={
+                                    'first_epoch_num': state.timer.epoch.value,
+                                    'epoch_count': 1
+                                })
+
+    def epoch_end(self, state: State, logger: Logger) -> None:
+        if rank_zero():
+            self.mllogger.event(key=constants.EPOCH_STOP, metadata={'epoch_num': state.timer.epoch.value})
+
+    def eval_start(self, state: State, logger: Logger) -> None:
+        if rank_zero():
+            self.mllogger.event(key=constants.EVAL_START, metadata={'epoch_num': state.timer.epoch.value})
+
+    def eval_end(self, state: State, logger: Logger) -> None:
+        if rank_zero():
+            accuracy = self._get_accuracy(state)
+
+            self.mllogger.event(key=constants.EVAL_STOP, metadata={'epoch_num': state.timer.epoch.value})
+            self.mllogger.event(key=constants.EVAL_ACCURACY,
+                                value=accuracy,
+                                metadata={'epoch_num': state.timer.epoch.value})
+            self.mllogger.event(key=constants.BLOCK_STOP, metadata={'first_epoch_num': state.timer.epoch.value})
+
+            if accuracy > self.target and not self.success:
+                self.mllogger.event(key=constants.RUN_STOP, metadata={"status": "success"})
+                self.mllogger.logger.removeHandler(self._file_handler)
+                self.success = True  # only log once
+
+
+def get_system_description(
+    submitter: str,
+    division: str,
+    status: str,
+    system_name: Optional[str] = None,
+) -> Dict[str, str]:
+    """Generates a valid system description.
+
+    Make a best effort to auto-populate some of the fields, but should
+    be manually checked prior to submission. The system name is
+    auto-generated as "[world_size]x[device_name]_composer", e.g.
+    "8xNVIDIA_A100_80GB_composer".
+
+    Args:
+        submitter (str): Name of the submitter organization
+        division (str): Submission division (open, closed)
+        status (str): system status (cloud, onprem, preview)
+
+    Returns:
+        system description as a dictionary
+    """
+    is_cuda = torch.cuda.is_available()
+    cpu_info = cpuinfo.get_cpu_info()
+
+    system_desc = {
+        "submitter": submitter,
+        "division": division,
+        "status": status,
+        "number_of_nodes": dist.get_world_size() / dist.get_local_world_size(),
+        "host_processors_per_node": "",
+        "host_processor_model_name": str(cpu_info.get('brand_raw', "CPU")),
+        "host_processor_core_count": str(psutil.cpu_count(logical=False)),
```

Comment:
Yes this would be incorrect on k8s, but runs will be occupy the entire node. This is meant to be best effort for the general case by users (see disclaimer) so need not be accurate in every circumstance.